### PR TITLE
fix(TDP-4084): lazy jackson parsing

### PR DESCRIPTION
**Pull request by hmdebenque**
- use JSonFactory.createParser(Reader) instead of the InputStream one
  to avoid read for type detection
- allow fetching only records in dataset content

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4084

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/180/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/180/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/186/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/186/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/190/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/190/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/194/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/194/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/202/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/202/)

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/api/action/context/ActionContext.java [hunk 1]
```diff
@@ -194,6 +194,11 @@ public boolean contains(String key) {
         return value;
     }
 
+    /** Remove context cached entry. */
+    public void evict(String key) {
```
Comment (elogeart) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/api/action/context/ActionContext.java]: How about a small unit test for this small method ?

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/TransformerWriter.java [hunk 2]
```diff
@@ -95,22 +95,12 @@ default void flush() throws IOException {
     }
 
     /**
-     * Close the data stream properly when all data is written or a stop signal is received. Not meant to close the OutputStream.
+     * Close the data stream properly when all data is written or a stop signal is received. Not meant to cloase the OutputStream.
```
Comment (elogeart) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/TransformerWriter.java]: Typo "cloase" -> "close"

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/221/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/221/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/226/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/226/)

File: dataprep-backend-common/src/test/java/org/talend/dataprep/transformation/api/action/context/ActionContextTest.java [hunk 3]
```diff
@@ -84,7 +88,34 @@ public void testColumnCache() throws Exception {
...
+        context.get(key, generationCode);
+
+        // Value is not generated again, cache works
+        assertFalse(valueGenerated.get());
+
+        context.evict(key);
+        context.get(key, generationCode);
+
+        // evict makes the generation work again
+        assertTrue(valueGenerated.get());
```
Comment (elogeart) [inline dataprep-backend-common/src/test/java/org/talend/dataprep/transformation/api/action/context/ActionContextTest.java]: I would have imagined a simpler test, based on the `contains(key)` method, how about this idea ?

**Pull request merged**